### DISCUSSION TITLE
Fuel calculations

### DISCUSSION THREAD

**Discussion by hudkinsnoah** [JSBSim-Team/jsbsim#1064] [Support requests]
I am trying to find how much fuel an aircraft has left in it's tank and have been using a few different methods with varying results. The first option was to take the total fuel capacity of a fuel tank using "propulsion/tank/contents-lbs". This I would assume gives me the weight of the fuel that I have currently in the tank. 

I then would multiply this by the percent full we have in the tank from "propulsion/tank/pct-full" divided by 100 so if for example we had a 120 gallon tank and 50 percent capacity it would look like this : 120 * (50 / 100). The last step is I would divide by the density of the lbs per gallon. This I would assume give me the current capacity of fuel I have in gallons. 

The issue with this, is I am seeing the fuel flow "propulsion/engine/fuel-flow-rate-gph" is giving me a number that is not being reflected by my formula. There are numbers at times of 800 gph. That should mean we are seeing a gallon lost every ~4.5 seconds. I am seeing that I am losing fuel at every ~9 seconds.

The second option that I have used to calculate fuel is taking that same fuel capacity from the first option "propulsion/tank/contents-lbs" and taking away fuel used "propulsion/engine/fuel-used-lbs". This is then divided by the fuel density "propulsion/tank/density-lbs_per_gal". The problem with this one, is I can see that when I look at pct-full even when I have calculated an empty tank, the value is still showing greater than 0. I haven't tried it more than 2 times since it takes a while to get down to 0 fuel by the calculation, but both times, the percent was in the 40's. 

I am not sure what I am doing wrong here to calculate fuel, but any help would be greatly appreciated, thanks!

**Comment by hudkinsnoah** [comment]
I think what I needed to actually do was take the tank contents and just divide that by the pounds density pounds per gallon. That seemed to make things line up with the fuel flow and with the pct-full

**Comment by seanmcleod** [comment]
> take the total fuel capacity of a fuel tank using "propulsion/tank/contents-lbs". 

I think you're mixing up capacity and contents. Here is an example from the 737 model.

```xml
        <tank type="FUEL">    <!-- Tank number 1 -->
            <location unit="IN">
                <x> 1327 </x>
                <y> 0 </y>
                <z> -69.57 </z>
            </location>
            <capacity unit="LBS"> 10912.8 </capacity>
            <contents unit="LBS"> 5456.4 </contents>
        </tank>
```

So it has the capacity to hold 10912lbs of fuel but it's contents is currently 5456lbs, so it's currently 50% full.
